user-guide/Advanced_Modules/User Defined APIs/UD_APIs_Triggering_an_API.md

```diff
@@ -0,0 +1,153 @@
+---
+uid: UD_APIs_Triggering_an_API
+---
+
+# Triggering a user-defined API
+
+> [!WARNING]
+> This feature is in preview and is not fully released yet. For now, it should only be used on a staging platform. It should not be used in a production environment.
+
+To trigger an API, you can send an HTTP or HTTPS request to the *UserDefinableApiEndpoint* DxM.
+
+> [!NOTE]
+>
+> - You can also trigger a user-defined API through the [SLNetClientTest Tool](xref:SLNetClientTest_triggering_api) for testing. However, note that you should always be extremely careful when using this tool, as it can have far-reaching consequences on the functionality of your DataMiner System.
+> - API triggers are handled asynchronously. To protect DataMiner, there is a limit to the number of concurrent triggers. As soon as that limit is reached, new triggers are added to a queue, to be handled as soon as another trigger is finished. It is not possible to adjust this limit, as it is automatically set based on the number of logical processors in the system (with a minimal concurrency of 4). The exact limit is logged in the file `C:\Skyline DataMiner\Logging\SLUserDefinableApiManager.txt`. Apart from this limit implemented by DataMiner, IIS for Windows 10 also has a concurrency limit of 10. IIS for Windows Server has no limits.
+
+## Building the HTTP request
+
+To test sending a request, you can use a tool like Postman.
+
+### URL
+
+To send a request to *UserDefinableApiEndpoint*, use a URL in the format `http(s)://{hostname}/api/custom/{route}`. (See [Route](xref:UD_APIs_Define_New_API#route).)
+
+Examples:
+
+```
+http://staging-agent-b3/api/custom/switchers/status
+https://production-agent-a1/api/custom/encoder/main/status
+```
+
+### HTTP methods
+
+The following HTTP methods are supported: GET, PUT, POST, DELETE.
+
+> [!NOTE]
+> You are free to choose which of these methods your API will support.
+
+### Authentication
+
+To authenticate yourself to the API, add a `Bearer` authorization header to your request containing the secret.
+
+Example of the header:
+
+```
+Bearer 24te8bRBOhrDj6TkXkim4qh9GpDXsdGudib57j7C/CU=
+```
+
+### Content-Type
+
+In case the API needs input from the user using the HTTP body, you need to specify a Content-Type header.
+
+The endpoint supports the following content types:
+
+- application/json
+- application/xml
+- text/xml
+- text/plain
+
+> [!NOTE]
+> Adding no Content-Type header to a request is also supported. In this case, the body is expected to be empty.
+
+### Content-Length
+
+The Content-Length header should be calculated and filled in automatically depending on how you send the request. It contains the length of the body in bytes. It is important that this is correct, as *UserDefinableApiEndpoint* will only read the number of bytes of the body specified in the Content-Length header.
```

Comment:
It should always be correctly filled in. If you use Postman, it will automatically calculate this based on the content. But you can manually override it. In the HttpClient in .NET, it will calculate this header, I don't think you can override it there. This contains probably too much information, I think the note above this section provides enough information about this header